action.yml

```diff
@@ -0,0 +1,295 @@
+    required: false
+    default: './suites'
+  TEST_SUITE_NAME:
+    description: 'which test suite to run'
+    required: true
+  WORKSPACE:
+    description: 'path to WORKSPACE'
+    required: false
+    default: './.leviathan-inputs'
+  DEVICE_TYPE:
+    description: "slug of device type"
+    required: true
+  REPORTS:
+    description: "path to reports directory"
+    required: false
+    default: "./reports"
+  ENVIRONMENT:
+    description: "What BalenaCloud env, balena-cloud or balena-machine"
+    required: true
+    default: "balena-cloud"
+  BALENACLOUD_ORG:
+    required: false
+    description: "Organization used during cloud suite to create fleets, etc."
+    default: "testbot"
+  BALENACLOUD_API_KEY:
+    required: false
+  BALENAMACHINE_API_KEY:
+    required: false
+  BALENACLOUD_APP_NAME:
+    description: "Testbot fleet for finding available Leviathan workers. Not used for QEMU workers. Can accept a list of apps separated by commas."
+    required: false
+    default: "balena/testbot-rig"
+  BALENACLOUD_API_URL:
+    required: false
+    default: "balena-cloud.com"
+  BALENAMACHINE_API_URL:
+    required: false
+    default: "bm.balena-dev.com"
+  BALENACLOUD_SSH_URL:
+    required: false
+    default: "ssh.devices.bm.balena-dev.com"
+  BALENACLOUD_SSH_PORT:
+    default: 222
+    required: false
+  QEMU_CPUS:
+    required: false
+    default: 4
+  QEMU_MEMORY:
+    required: false
+    default: "1G"
+  WORKER_TYPE:
+    description: "Valid choices: testbot/qemu"
+    required: true
+  LEVIATHAN_DEBUG:
+    description: |
+      Use this parameter to modify the default debugging settings on test runs. To debug test runs when rebuilding, add new values to this parameter separated with commas. Know more about the possible values available for debugging: https://balena-os.github.io/leviathan/pages/Getting-Started/debugging.html Example value: dev: true, failFast: false
+    required: false
+
+
+outputs:
+  reports:
+    description: "REPORT!"
+    value: ${{ steps.random-number-generator.outputs.random-number }}
+
+runs:
+  using: "composite"
+  steps:
+    # - name: Past Cleanup probably not required
+    #   shell: bash
+    #   run: |
+    #     # unintended consequences
+    #     # It does delete a ton of images on GHR: 
+    #     # Total reclaimed space: 3.141GB
+    #     # docker system prune -a -f || true
+    #     printenv | sort
+
+    - name: Checkout the current repo
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Fetching the right leviathan
+      uses: actions/checkout@v4
+      with:
+        repository: balena-os/leviathan
+        submodules: true
+        path: ./leviathan-workspace
+        # The branch, tag or SHA to checkout. When checking out the repository that
+        # triggered a workflow, this defaults to the reference or SHA for that event.
+        # Otherwise, uses the default branch.
+        ref: ${{ inputs.LEVIATHAN_REF }}
```

Comment:
Same here, composite actions don't need to perform a checkout. The calling workflow should do that.